en/scripting/setup.md

```diff
@@ -77,75 +72,159 @@ Before writing code, please read the following documentations to learn more abou
+- The above directory can be created manually. You can also use the menu in the right-click menu of **Asset** to generate a directory after clicking.
+![custom script](setup/custom-script.png)
 
-```TypeScript
+The default `NewComponent` script template is still in the engine built-in resources directory `resources\3d\engine\editor\assets\default_file_content\ts`.
+The code of the file is below:
+
+```ts
 import { _decorator, Component, Node } from 'cc';
 const { ccclass, property } = _decorator;
 
-@ccclass('Hello')
-export class Hello extends Component {}
+@ccclass('<%UnderscoreCaseClassName%>')
+export class <%UnderscoreCaseClassName%> extends Component {
+    start() {
+
+    }
+
+    update(deltaTime: number) {
+        
+    }
+}
+
+/**
+ * COMMENTS_GENERATE_IGNORE
+ * Use "COMMENTS_GENERATE_IGNORE" tag if you do not want later created scripts to contain these comments.
+ * 
+ * Predefined Variables
+ * You can use predefined variables below to setup your scripting preference. For example, whether to use camel case style.
+ * 
+ * <%UnderscoreCaseClassName%>, class name in underscore format, like 'new_component'
+ * <%CamelCaseClassName%>, class name in camel format, like 'NewComponent'
+ * <%Author%>, Who create this file
+ * <%DateTime%>, when create this file
+ * <%FileBasename%>, creating file name with extension
+ * <%FileBasenameNoExtension%>, creating file name without extension
+ * <%URL%>, url of this file in COCOS ASSET URL format
+ * <%ManualUrl%>, url of office help document, like 'https://docs.cocos.com/creator/manual/en/'
+ *
+ * 
+ * Example:
+ * 
+  @ccclass('<%UnderscoreCaseClassName%>')
+  export class <%UnderscoreCaseClassName%> extends Component {
+  
+    // class member could be defined like this.
+    dummy = '';
+
+    // Use 'property' decorator if your want the member to be serializable.
+    @property
+    serializableDummy = 0;
+
+    start () {
+        // Your initialization goes here.
+    }
+
+    update (deltaTime: number) {
+        // Your update function goes here.
+    }
+
+  }
+ *
+ * Learn more about scripting: <%ManualUrl%>scripting/
+ * Learn more about CCClass: <%ManualUrl%>scripting/decorator.html
+ * Learn more about life-cycle callbacks: <%ManualUrl%>scripting/life-cycle-callbacks.html
+ */
+
 ```
 
-After saving the script and returning to the editor, notice that the script component name in the **Inspector** panel has changed to **Hello**, but the script file name is still the original name, `say-hello`:
+**Nodes**
+
+> A large number of comments in the script template will not be generated into the script file, because we use the keyword annotation `COMMENTS_GENERATE_IGNORE` in the comment. As long as this keyword is in a comment, the generated script file will ignore the comment Lose.
+
+> `Predefined Variables` We have prepared some pre-made variables, which can be used as auxiliary information when generating script files, such as author `<%Author%>`.
```

Comment:
```suggestion
> **Nodes**:
>
> 1. A large number of comments in the script template will not be generated into the script file, because we use the keyword annotation `COMMENTS_GENERATE_IGNORE` in the comment. As long as this keyword is in a comment, the generated script file will ignore the comment Lose.
>
> 2. `Predefined Variables` We have prepared some pre-made variables, which can be used as auxiliary information when generating script files, such as author `<%Author%>`.
```